src/libraries/System.Net.Sockets/src/System/Net/Sockets/Socket.Tasks.cs

```diff
@@ -98,17 +127,17 @@ internal Task ConnectAsync(IPAddress[] addresses, int port)
                 throw new ArgumentException(SR.net_invalidAddressList, nameof(addresses));
             }
 
-            return DoConnectAsync(addresses, port);
+            return DoConnectAsync(addresses, port, cancellationToken);
         }
 
-        private async Task DoConnectAsync(IPAddress[] addresses, int port)
+        private async ValueTask DoConnectAsync(IPAddress[] addresses, int port, CancellationToken cancellationToken)
         {
             Exception? lastException = null;
             foreach (IPAddress address in addresses)
             {
                 try
                 {
-                    await ConnectAsync(address, port).ConfigureAwait(false);
+                    await ConnectAsync(address, port, cancellationToken).ConfigureAwait(false);
```

Comment:
If cancellation is requested, the catch block is going to catch it, and then it's going to try the next address (and be canceled, and then try the next address, and be canceled, etc).  Seems like the catch block should be changed to be:
```C#
catch (Exception ex) when (ex is not OperationCanceledException)
```
or something like that.

Reply:
Yep, good point, will fix